### DISCUSSION TITLE
Volume knob doesn't move sometimes

### DISCUSSION THREAD

**Discussion by subatomicglue** [rern/rAudio#614] [Fixed Issues]
Using: RPi 4b on 7" touchscreen with 20211112

When rotating the volume knob, sometimes it wont move fluidly.  Sometimes it does.
- I think that sometimes the touch is inaccurate because fingers are fat - and misses the blue dot.   Resulting in hard to move volume knob.     
- It'd be easier to use, higher chance of success of rotating it, if "mouse/touch down" event followed by "mouse/touch move" event moves outside the knob's radius, then the knob becomes locked on to the finger's position around the circle.  

See video for the UI bug/problem:
https://user-images.githubusercontent.com/7660853/141527999-7c13cfab-1768-47d7-8651-628d532b2c1c.MOV

**Comment by cYCL157** [comment]
When i use the touchscreen on mine, I tend to not try to use the "knob" to move, but touch the final destination I want it at.  My fingers may or may not be too fat......

**Reply by subatomicglue** [reply]
For sure that is _also_ an interaction that should exist

But, grabbing the blue dot right now is too hard and too easy to miss.  And I'd like that improved.  

Without losing the single touch to set functionality, of course. ;-)

**Comment by rern** [comment]
Local touchscreens in general are not suitable for use as primary touch interface. It's the hardware issue.

**Reply by subatomicglue** [reply]
Though I empathize that maybe there is an architectural issue with this tech stack.    Touch screen definitely needs to be usable, ideally  ;-).  

At least, for me, no one is using their smart phone with this, too inconvenient, awkward, hard to connect, no QR code when things are playing so it's unfindable, etc.  this is meant to be a kiosk in our kitchen basically, family access...

if you can point me at the code for the volume control, I wouldn't mind taking a look... thanks

**Reply by ceejayemm** [reply]
RPI3A+(headless with BT Speaker) and RPI3B+ (Touchscreen and Merus Amp, Q Associates Q3020i speakers), both have RA i20211022 release patched up currently to 20211203 update version.

I would agree. The touch screen generally works well but it has at least a couple of issues. The (slightly) annoying thing is that most of these issues have worked in the past but seems to be being left behind because of 'tech issues' that don't currently seem fixable.  Like @subatomicglue, I use my rAudio-1 installation as a kiosk and it has to be easy to use for my wife and other family members. Having to resort to a PC/web app to access (some parts) of a sound system like this is counter-productive for non-techies and I am running out of explanations when another function becomes not available.

For me the issues are the inability to access the top right cog menu (setup), and not being able to exit any screen where a vertical scrollbar becomes visible both reported in #426. Also for completeness, but not necessarily related to the touchscreen issues, the system drops in to 'stop' mode when a BT speaker is closed (#595) but does not reconnect when the speaker is re-opened. This last issue again means that a PC/web app is needed to restart the audio playing. None of these issues are present in a PC web browser (Chrome / MS Edge) session nor when using the Android app.

Don't get me wrong, rAudio-1 is a great product which I have used for a number of years (in its many previous guises) and I really like how it works and the sound quality is excellent (to me).  @rern provides excellent support and responses, if we could solve the few glitches then it would be really great. As always I am happy to test ant possible release prior to general release.

Many thanks

Chris
